src/main/java/sg/ncl/MainController.java

```diff
@@ -2607,16 +2607,30 @@ public String adminDataManagement(Model model, HttpSession session) {
             datasetsList.add(dataset);
         }
 
-        ResponseEntity response4 = restTemplate.exchange(properties.getDownloadStat(), HttpMethod.GET, request, String.class);
-        String responseBody4 = response4.getBody().toString();
+        response = restTemplate.exchange(properties.getDownloadStat(), HttpMethod.GET, request, String.class);
+        responseBody = response.getBody().toString();
 
         Map<Integer, Long> dataDownloadStats = new HashMap<>();
-        JSONArray statJsonArray = new JSONArray(responseBody4);
-        for (int i = 0; i < statJsonArray.length(); i++) {
-            JSONObject statInfoObject = statJsonArray.getJSONObject(i);
+        JSONArray statJsonArray1 = new JSONArray(responseBody);
+        for (int i = 0; i < statJsonArray1.length(); i++) {
+            JSONObject statInfoObject = statJsonArray1.getJSONObject(i);
             dataDownloadStats.put(statInfoObject.getInt("dataId"), statInfoObject.getLong("count"));
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "dataId" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "count" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)